python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+        """Make an `lsst.afw.geom.SkyWcs` from a dictionary of mappings
+
+        Parameters
+        ----------
+        mapDict : `dict`
+            Dictionary of mapping parameters
+        centerRA : `lsst.geom.Angle`
+            RA of the tangent point
+        centerDec : `lsst.geom.Angle`
+            Declination of the tangent point
+        doNormalizePixels : `bool`
+            Whether to normalize pixels so that range is [-1,1]
+        xScale : `float`
+            Factor by which to normalize x-dimension. Corresponds to width of
+            detector
+        yScale : `float`
+            Factor by which to normalize y-dimension. Corresponds to height of
+            detector
+
+        Returns
+        -------
+        outWCS : `lsst.afw.geom.SkyWcs`
+            WCS constructed from the input mappings
+        """
+        # Set up pixel frames
+        pixelFrame = astshim.Frame(2, "Domain=PIXELS")
+        normedPixelFrame = astshim.Frame(2, "Domain=NORMEDPIXELS")
+
+        if doNormalizePixels:
+            # Pixels will need to be rescaled before going into the mappings
+            normCoefficients = [-1.0, 2.0/xScale, 0,
+                                -1.0, 0, 2.0/yScale]
+            normMap = convertToAstPolyMapCoefficients(normCoefficients)
+        else:
+            normMap = astshim.UnitMap(2)
+
+        # All of the detectors for one visit map to the same tangent plane
+        tangentPoint = lsst.geom.SpherePoint(centerRA, centerDec)
+        cdMatrix = afwgeom.makeCdMatrix(1.0 * lsst.geom.degrees, 0 * lsst.geom.degrees, True)
+        iwcToSkyWcs = afwgeom.makeSkyWcs(lsst.geom.Point2D(0, 0), tangentPoint, cdMatrix)
+        iwcToSkyMap = iwcToSkyWcs.getFrameDict().getMapping("PIXELS", "SKY")
+        skyFrame = iwcToSkyWcs.getFrameDict().getFrame("SKY")
+
+        frameDict = astshim.FrameDict(pixelFrame)
+        frameDict.addFrame("PIXELS", normMap, normedPixelFrame)
+
+        currentFrameName = "NORMEDPIXELS"
+
+        # Assume python 3.7+ so dict is ordered:
+        for m, mapElement in enumerate(mapDict.values()):
+            mapType = mapElement['Type']
+
+            if mapType == 'Poly':
+                mapCoefficients = mapElement['Coefficients']
+                astMap = convertToAstPolyMapCoefficients(mapCoefficients)
+            elif mapType == 'Identity':
+                astMap = astshim.UnitMap(2)
+            else:
+                raise ValueError(f"Converting map type {mapType} to WCS is not supported")
+
+            if m == len(mapDict) - 1:
+                newFrameName = "IWC"
+            else:
+                newFrameName = "INTERMEDIATE" + str(m)
+            newFrame = astshim.Frame(2, f"Domain={newFrameName}")
+            frameDict.addFrame(currentFrameName, astMap, newFrame)
+            currentFrameName = newFrameName
+        frameDict.addFrame("IWC", iwcToSkyMap, skyFrame)
+
+        outWCS = afwgeom.SkyWcs(frameDict)
```

Comment:
If you're going to use camelCase throughout the code, this should be `outWcs`. I would suggest switching to snake_case throughout, though (see [RFC-623](https://jira.lsstcorp.org/browse/RFC-623)).

Reply:
First of all, I agree with using snake_case for new code.  However, the LSST dev guide is WRONG on this front re: camelCase, see PEP-8: https://peps.python.org/pep-0008/ "Note: When using acronyms in CapWords, capitalize all the letters of the acronym. Thus HTTPServerError is better than HttpServerError."

Reply:
Ok, I'm not changing all the internal variables, but I made the API all snake_case.